air/src/trace/main_trace.rs

```diff
@@ -367,6 +357,30 @@ impl MainTrace {
         self.columns.get_column(MEMORY_V_COL_RANGE.start + 3)[i]
     }
 
+    /// Computes a memory read request at `row` given randomness `alphas`, memory address `addr`
+    /// and value `value`.
+    pub fn compute_memory_read_request<E: FieldElement<BaseField = Felt>>(
+        &self,
+        alphas: &[E],
+        row: usize,
+        addr: Felt,
+        value: Word,
+    ) -> E {
+        let op_label = MEMORY_READ_LABEL;
+        let ctx = self.ctx(row);
+        let clk = self.clk(row);
+
+        alphas[0]
```

Comment:
Just a note: with #1223 we would take `alphas[0]` out of each memory request (such that each request is truly a random linear combination as opposed to an affine one). But we should leave as is in this PR, and change it only in #1223.